### PR TITLE
Fix prior scales

### PR DESCRIPTION
**Pull request by yannikschaelte**
* fix get_prior_from_df returning wrong scales if using default bounds
* fix get_prior_from_df handling missing entries badly
* add tests

### CODE REVIEW

File: tests/test_lint.py [hunk 1]
```diff
@@ -258,13 +258,30 @@ def test_check_parameter_bounds():
...
+         OBJECTIVE_PRIOR_TYPE: [NORMAL, LOG_NORMAL, '']}))
     lint.assert_parameter_prior_type_is_valid(pd.DataFrame())
 
     with pytest.raises(AssertionError):
         lint.assert_parameter_prior_type_is_valid(pd.DataFrame(
-            {INITIALIZATION_PRIOR_TYPE: ['normal', '']}))
+            {INITIALIZATION_PRIOR_TYPE: ['normel']}))
+
+
+def test_assert_parameter_prior_parameters_are_valid():
```
Comment (dweindl) [inline tests/test_lint.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4984605)

File: petab/lint.py [hunk 2]
```diff
@@ -404,15 +404,62 @@ def assert_parameter_prior_type_is_valid(
...
+                # empty parameters fine
+                continue
+            # parse parameters
+            try:
+                pars = tuple([float(val) for val in pars_str.split(';')])
+            except ValueError:
+                raise AssertionError(
+                    f"Could not parse prior parameters '{pars}'.")
+            # all distributions take 2 parameters
+            if 2 != len(pars):
```
Comment (dweindl) [inline petab/lint.py]: Codacy Issue found: [Comparison should be len(pars) != 2](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4984605)

File: tests/test_petab.py [hunk 3]
```diff
@@ -164,10 +164,48 @@ def test_get_observable_id():
     assert petab.get_observable_id('sigma_obs1') == 'obs1'
 
 
+def test_get_priors_from_df():
```
Comment (dweindl) [inline tests/test_petab.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4984605)

File: petab/core.py [hunk 4]
```diff
@@ -251,6 +251,18 @@ def to_float_if_float(x: Any) -> Any:
         return x
 
 
+def is_empty(val) -> bool:
```
Comment (dweindl) [inline petab/core.py]: Isn't that pretty much https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.isnull.html?

File: petab/lint.py [hunk 5]
```diff
@@ -404,15 +404,62 @@ def assert_parameter_prior_type_is_valid(
...
+                raise AssertionError(
+                    f"The prior parameters '{pars}' do not contain the "
+                    "expected number of entries.")
```
Comment (dweindl) [inline petab/lint.py]: Shall we be more friendly and print expected type/number of parameters?

File: tests/test_lint.py [hunk 6]
```diff
@@ -257,14 +257,34 @@ def test_check_parameter_bounds():
...
-        {INITIALIZATION_PRIOR_TYPE: [UNIFORM, LAPLACE],
-         OBJECTIVE_PRIOR_TYPE: [NORMAL, LOG_NORMAL]}))
+        {INITIALIZATION_PRIOR_TYPE: [UNIFORM, LAPLACE, ''],
+         OBJECTIVE_PRIOR_TYPE: [NORMAL, LOG_NORMAL, '']}))
     lint.assert_parameter_prior_type_is_valid(pd.DataFrame())
 
     with pytest.raises(AssertionError):
         lint.assert_parameter_prior_type_is_valid(pd.DataFrame(
-            {INITIALIZATION_PRIOR_TYPE: ['normal', '']}))
+            {INITIALIZATION_PRIOR_TYPE: ['normel']}))
```
Comment (erikadudki) [inline tests/test_lint.py]: 'normal' instead of 'normel' ?

File: petab/core.py (hunk 4 above)
Comment (yannikschaelte) [reply petab/core.py]: just checked: ``pd.isnull('') is False``, for None and nan we have True however. Maybe have '' or pd.isnull in the function?

File: tests/test_lint.py (hunk 6 above)
Comment (yannikschaelte) [reply tests/test_lint.py]: nonono that pun was intended ;) ;) ;)

File: petab/core.py (hunk 4 above)
Comment (dweindl) [reply petab/core.py]: Okay, then either way is fine

File: tests/test_petab.py [hunk 7]
```diff
@@ -164,6 +164,43 @@ def test_get_observable_id():
...
```
Comment (dweindl) [inline tests/test_petab.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4984605)

File: petab/core.py (hunk 4 above)
Comment (yannikschaelte) [reply petab/core.py]: added pd.isnull now. could check for more eventualities, so better

**Pull request merged**